### PR TITLE
sys-apps/man-db: Add systemd timer support

### PR DESCRIPTION
**Pull request by comio**
Should solve [622236](https://bugs.gentoo.org/show_bug.cgi?id=622236)

ciao

luigi

### CODE REVIEW

File: sys-apps/man-db/man-db-2.7.6.1-r3.ebuild [hunk 1]
```diff
@@ -12,7 +12,7 @@ SRC_URI="mirror://nongnu/${PN}/${P}.tar.xz"
 LICENSE="GPL-3"
 SLOT="0"
 KEYWORDS="alpha amd64 arm ~arm64 hppa ia64 ~m68k ~mips ppc ppc64 ~s390 ~sh sparc x86 ~amd64-linux ~arm-linux ~x86-linux"
-IUSE="berkdb +gdbm +manpager nls selinux static-libs zlib"
+IUSE="berkdb +cron +gdbm +manpager nls selinux static-libs systemd zlib"
```
Comment (Whissi) [inline sys-apps/man-db/man-db-2.7.6.1-r3.ebuild]: No, we don't use USE flags to control the installation of small files.
People who don't want such small files on their systems should make use of portage's *INSTALL_MASK* feature.

File: sys-apps/man-db/man-db-2.7.6.1-r3.ebuild [hunk 2]
```diff
@@ -105,4 +113,9 @@ pkg_postinst() {
 		einfo "Rebuilding man-db from scratch with new database format!"
 		mandb --quiet --create
 	fi
+
+	if use systemd; then
```
Comment (Whissi) [inline sys-apps/man-db/man-db-2.7.6.1-r3.ebuild]: That's not what [Mike's comment](https://bugs.gentoo.org/show_bug.cgi?id=622236#c1) was about.

File: sys-apps/man-db/man-db-2.7.6.1-r3.ebuild (hunk 1 above)
Comment (comio) [inline sys-apps/man-db/man-db-2.7.6.1-r3.ebuild]: The problem is not the installation of a small file. If I'm using an still mixed cron/systemd-timer system the installation of the cron files will have the side effect to automatically enable twice the activities: one from cron and one from systemd-timer. If I don't want cron files I should avoid to install these files.

File: sys-apps/man-db/man-db-2.7.6.1-r3.ebuild (hunk 2 above)
Comment (comio) [inline sys-apps/man-db/man-db-2.7.6.1-r3.ebuild]: I know but I prefer to show the message even if the systemd is not enabled yet.

File: sys-apps/man-db/man-db-2.7.6.1-r3.ebuild (hunk 1 above)
Comment (Whissi) [inline sys-apps/man-db/man-db-2.7.6.1-r3.ebuild]: Wait, the systemd timer won't be enabled automatically. The message shown in *pkg_postinst()* might tell you about the possibility to use a systemd timer instead of a cron job. So if you want to use a systemd timer instead of a cron job you are supposed to remove the default cron job.

At the moment, if you don't like an ebuild default, i.e. an automatically installed cron job, you are supposed to add an *INSTALL_MASK* for that particular file.

File: sys-apps/man-db/man-db-2.7.6.1-r3.ebuild (hunk 2 above)
Comment (Whissi) [inline sys-apps/man-db/man-db-2.7.6.1-r3.ebuild]: I respect your preferences but when Gentoo added support for systemd we explicit voted against the use of a systemd USE flag to control such things. So at the moment, a contribution adding a systemd USE flag for things like that has no chance of getting merged.

**Comment by Whissi** [comment]
A general note:

The problem raised in [Gentoo Bug #622236](https://bugs.gentoo.org/show_bug.cgi?id=622236) is a challenging problem: At the moment, cron jobs are the default and they don't hurt, because we don't install any cron implementation per default.

If anyone really wants to get rid of cron in favor of systemd timers (and I don't know if Gentoo wants to support something like that, at least at the moment), you cannot expect a smooth transition phase. It also doesn't really work if we will receive such contribution in future for every package installing a cron job. Yes, people who wants to replace cron with systemd timers probably might want something like that but we would need to find a solution which works for the others who wants to keep cron as well in general.

tl;dr

A base-system package is probably not the right place to start something like that. Start working with Gentoo's systemd project for a proposal how to deal with systemd timers. Once such a proposal was accepted by the community we can start migrating packages, including base-system packages.

File: sys-apps/man-db/man-db-2.7.6.1-r3.ebuild (hunk 2 above)
Comment (comio) [inline sys-apps/man-db/man-db-2.7.6.1-r3.ebuild]: ok.

**Comment by comio** [comment]
@Whissi,

your general note is very interesting point of view (I read even if the tl;dr warning).
My proposals are just a small contribute to solve some day-by-day issue. In my case I have double calls to tools, one from cron job script and one from the systemd timer.

I started looking another package (logrotate) in order to see how others solved the issue. I looked to wrong place...

At the end, I'm a little bit confusing how we can give contribution.

ciao

luigi

**Comment by Whissi** [comment]
> My proposals are just a small contribute to solve some day-by-day issue.

I understand. But this is a bigger problem which needs a general solution and base-system is not the place to try new things. So if you are an early adopter and using new things you are supposed to deal with such problems. But adding an *INSTALL_MASK* should fix your double call problem (and don't forget: You only have a double call problem because you manually added a systemd timer but didn't remove the cron job).

**Comment by floppym** [comment]
> I understand. But this is a bigger problem which needs a general solution and base-system is not the place to try new things. 

I don't see why you are making this a "big problem". Installing systemd timers as an alternative to cron files seems like an obvious thing to me.

If you would prefer to have the systemd team commit them, we can probably do that.

**Comment by Whissi** [comment]
> I don't see why you are making this a "big problem". Installing systemd timers as an alternative to cron files seems like an obvious thing to me.

Yes, just installing an additional systemd timer is fine. But this was originally about something else: @comio has a system with both, a cron daemon **and** systemd installed. He now wanted to use systemd timers but he didn't want to manually remove the cron job or was trying to do control the cron job installation via USE flag.

**Comment by floppym** [comment]
> Yes, just installing an additional systemd timer is fine. But this was originally about something else: @comio has a system with both, a cron daemon and systemd installed. He now wanted to use systemd timers but he didn't want to manually remove the cron job or was trying to do control the cron job installation via USE flag.

Ah, I see. I think you have the right idea: just install the timer unit, and let the sysadmin deal with enabling it and possibly disabling the cron job if a cron daemon is running.

**Comment by comio** [comment]
The ebuild should avoid to install not useful things  or, at least inform about the side effects. The real issue is that just coping inside cron's directories is a bad solution because you haven't the control about which scripts will be called. An eselect based solution should mitigate the issue, permitting you to have fine grain control regarding the enabled cronjobs. 

Ciao

Luigi

**Comment by flappyports** [comment]
no activity
